supervisor/backups/manager.py

```diff
@@ -366,19 +387,24 @@ async def _do_restore(
             return False
         else:
             if addon_start_tasks:
+                change_stage(RestoreJobStage.AWAIT_ADDON_RESTARTS)
                 # Ignore exceptions from waiting for addon startup, addon errors handled elsewhere
                 await asyncio.gather(*addon_start_tasks, return_exceptions=True)
 
             return True
         finally:
-            # Do we need start Home Assistant Core?
-            if not await self.sys_homeassistant.core.is_running():
-                await self.sys_homeassistant.core.start()
+            # Leave Home Assistant alone if it wasn't part of the restore
+            if homeassistant:
+                change_stage(RestoreJobStage.CHECK_HOME_ASSISTANT)
+
+                # Do we need start Home Assistant Core?
+                if not await self.sys_homeassistant.core.is_running():
+                    await self.sys_homeassistant.core.start()
```

Comment:
Tbh this entire section is an edge case IMO. When you actually restore Home Assistant it stops and then starts it. And that includes a lot of logic to check and make sure its actually running after that and rollback to the last known good version on failure.

So really there's only two times this code would do anything:
1. If HA restore totally failed (rollback too). Then this will be one last attempt to get it started again (which we probably don't have high hopes for at that point). And there's already been a ton of logging to this point so the user knows something has gone quite wrong
2. The user had stopped core manually (like an `ha core stop` in the CLI) and then restored a backup including home assistant via the cli (not via the UI). If HA was stopped manually by the user before restore began then the earlier steps won't try to start HA but this one will. This is definitely an edge case, restores are generally done via the UI (so HA is definitely running) or without stopping HA first.

So I don't think any more detail is required here. This will rarely come up. The logging would only matter in (2) above and I'm not even fully convinced we should be starting HA again in that edge case. I mentioned that earlier in my comment.